proxy/src/main/java/com/velocitypowered/proxy/plugin/VelocityPluginManager.java

```diff
@@ -92,19 +100,44 @@ public void loadPlugins(Path directory) throws IOException {
         }
       }
 
-      // Actually create the plugin
-      PluginContainer pluginObject;
+      try {
+        VelocityPluginContainer container = new VelocityPluginContainer(plugin);
+        pluginContainers.put(container, loader.createModule(container));
+      } catch (Exception e) {
+        logger.error("Can't create module for plugin {}", plugin.getId(), e);
+      }
+    }
+
+    // Make a global Guice module that with common bindings for every plugin
+    AbstractModule commonModule = new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(ProxyServer.class).toInstance(server);
+        bind(PluginManager.class).toInstance(server.getPluginManager());
+        bind(EventManager.class).toInstance(server.getEventManager());
+        bind(CommandManager.class).toInstance(server.getCommandManager());
+        for (PluginContainer container : pluginContainers.keySet()) {
+          bind(PluginContainer.class)
```

Comment:
If you try to inject a `PluginContainer` for some other plugin but that plugin doesn't exist, Guice will refuse to initialize the object for the plugin. Given that Velocity also supports optional dependencies, this isn't some strange thing that will never happen - it is a possibility.

Reply:
Guice supports optional dependencies with
```java
@Inject(optional = true)
```